# Studentnotebook – Data Understanding
In dit werkcollege ga je werken met twee databronnen: een CSV-bestand en een SQLite-database. Je voert inspecties uit, controleert de kwaliteit van de data en vergelijkt beide bronnen.

**Fase binnen CRISP-DM:** Data Understanding – Describe & Verify

## 1. Business Understanding
- Wat is het doel van deze analyse?
- Welke inzichten zou je uiteindelijk willen opleveren?
- Noteer kort wat jouw verwachtingen zijn van de dataset.

*Plaats jouw uitwerking hier*

## 2. Data Understanding

### 2.1 Inspecteer de CSV uit de entreeticket

In [5]:
import pandas as pd

df_csv = pd.read_csv('klantdata.csv')
df_csv.head()

,klant_id,leeftijd,geslacht,totaal_bestellingen,totaal_retouren
0,1040,39.0,M,6,4.0
1,1099,33.0,Man,10,8.0
2,1083,41.0,Anders,4,4.0
3,1017,50.0,V,4,2.0
4,1057,32.0,vrouw,4,2.0


*Wat zie je? Wat zijn je eerste indrukken van de data? Ziet het er bruikbaar uit?*

ik zie tabellen:
- klant_id
- leeftijd
- geslacht
- totaal_bestellingen
- totaal_retouren

Mijn eerste indruk op de data is dat geslacht verschillende termen zoals Man en M. De data dat vertoond word is niet bruikbaar.

In [6]:
df_csv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 5 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   klant_id             120 non-null    int64  
 1   leeftijd             119 non-null    float64
 2   geslacht             120 non-null    object 
 3   totaal_bestellingen  120 non-null    int64  
 4   totaal_retouren      119 non-null    float64
dtypes: float64(2), int64(2), object(1)
memory usage: 4.8+ KB


*Wat zie je? Hoeveel regels aan data zijn er? Welke kolommen zijn er? Welke datatypes? Missende waarden?*

Er zijn 120 regels aan data en ik zie 5 kolommen met de datatypes int64, float64 en object. De tabellen leeftijd en totaal_retouren hebben beide 1 missende waarde.

In [7]:
df_csv.describe()

,klant_id,leeftijd,totaal_bestellingen,totaal_retouren
count,120.000000,119.000000,120.000000,119.000000
mean,1059.825000,33.915966,14.658333,13.823529
std,32.981702,12.530356,91.602493,91.991032
min,1001.000000,4.000000,0.000000,-1.000000
25%,1031.750000,28.000000,4.000000,2.000000
50%,1061.500000,34.000000,5.000000,4.000000
75%,1088.250000,38.500000,7.000000,6.000000
max,1115.000000,118.000000,999.000000,998.000000


*Wat zie je? Zijn er mogelijke onlogische waarden of verdelingen?*

Voor leeftijd is de standardafwijking van leeftijd is vrij groot vergeleken met de gemiddelde leeftijd.de 25, 50 en 75% kwartielen zit tussen de 28 en 38.5.  De kleinste waarde is 4 en hoogste waarde is 118. voor de totaal_bestellingen heeft een heel erg hoge standardafwijking in vergeleken met het gemiddelde. De kleinste waarde hiervan is 0 en hoogste is 999. de totaal retouren heeft ook een erg hoge standard afwijking van de gemiddelde waarden. de minimale waarden hiervan is -1 en de hoogste waarde is 998.

**Leeftijd**

De data voor leeftijd laat onlogische waarden zien die de gemiddelde beinvloeden.
- Bevinding: 75% van de klanten tussen de 28 en 38,5 jaar oud is, zijn de uiterste een probleem in de data.
- Verwoording: De tabel leeftijd bevat onrealistische waarden van 4 jaar en 118 jaar. Sinds de kwartielen (25%-75%) tussen de 28 39 jaar zitten, lijken deze invoerfouten zijn of uitschieters die de standaardafwijking onnodig verhogen.


### Totaal Bestellingen
De data is niet logsich door 1 of 2 waarden.
- De bevinding: Het gemiddelde is 14.6 maar de standaardafwijking is 91.6. Dit komt door de max van 999.


### 2.2 Inspecteer de SQLite-database

In [12]:
import sqlite3

conn = sqlite3.connect('klanten.db')
df_sql = pd.read_sql_query("SELECT * FROM klanten_data", conn)
conn.close()


In [13]:
df_sql.head()

,klant_id,geslacht,email_nieuwsbrief,account_jaar,productcategorie,aankoopdatum,verzend_datum,bedrag
0,1040,M,nee,2019,Kleding,2022-12-21 00:00:00,2022-12-30 00:00:00,57.22
1,1099,Man,ja,2019,Wonen,2021-12-04 00:00:00,2021-12-08 00:00:00,20.82
2,1083,Anders,nee,2020,Boeken,2023-05-29 00:00:00,2023-05-30 00:00:00,15.10
3,1017,V,ja,2021,Kleding,2023-12-13 00:00:00,2023-12-15 00:00:00,34.26
4,1057,vrouw,nee,2015,Kleding,2023-03-03 00:00:00,2023-03-04 00:00:00,59.03


*Wat zie je? Wat zijn je eerste indrukken van de data? Ziet het er bruikbaar uit?*

In [10]:
df_sql.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   klant_id           120 non-null    int64  
 1   geslacht           120 non-null    object 
 2   email_nieuwsbrief  120 non-null    object 
 3   account_jaar       120 non-null    int64  
 4   productcategorie   120 non-null    object 
 5   aankoopdatum       120 non-null    object 
 6   verzend_datum      108 non-null    object 
 7   bedrag             120 non-null    float64
dtypes: float64(1), int64(2), object(5)
memory usage: 7.6+ KB


*Wat zie je? Hoeveel regels aan data zijn er? Welke kolommen zijn er? Welke datatypes? Missende waarden? Verwacht je dat deze dataset te combineren is met de eerste dataset?*

## 3. Verify Data Quality
Voer hieronder je checks uit op je database dataset:

In [ ]:
# Missende waarden

In [ ]:
# Duplicaten

In [ ]:
# Aantal unieke categorieën

## 4. Combineer (merge) CSV en database op klant_id

In [ ]:
# voeg datasets samen

In [ ]:
# bekijk info van gecombineerde dataset

## 5. Kwaliteitscontrole op gecombineerde dataset

In [ ]:
# bekijk beschrijvende statistieken

In [ ]:
# missende waarden

In [ ]:
# duplicaten

In [ ]:
# andere inspecties

**Vragen:**
- Wat zie je nu dat je eerder niet zag?
- Is de data klaar voor analyse?

*Vul hier je antwoord in*

## 6. Reflectie
- Welke kwaliteitsproblemen heb je gevonden?
- Welke kolommen zou je opschonen of verwijderen?
- Wat zou je rapporteren aan een opdrachtgever?
- Wat neem je mee naar de volgende stap (Data Preparation)?